crates/turbopack-css/src/process.rs

```diff
@@ -0,0 +1,747 @@
+pub enum ParseCssResult {
+    Ok {
+        #[turbo_tasks(debug_ignore, trace_ignore)]
+        cm: Arc<swc_core::common::SourceMap>,
+
+        #[turbo_tasks(trace_ignore)]
+        stylesheet: StyleSheetLike<'static, 'static>,
+
+        references: Vc<ModuleReferences>,
+
+        url_references: Vc<UnresolvedUrlReferences>,
+
+        #[turbo_tasks(trace_ignore)]
+        options: ParserOptions<'static, 'static>,
+    },
+    Unparseable,
+    NotFound,
+}
+
+impl PartialEq for ParseCssResult {
+    fn eq(&self, _: &Self) -> bool {
+        false
+    }
+}
+
+#[turbo_tasks::value(shared, serialization = "none", eq = "manual")]
+pub enum CssWithPlaceholderResult {
+    Ok {
+        #[turbo_tasks(debug_ignore, trace_ignore)]
+        cm: Arc<swc_core::common::SourceMap>,
+
+        #[turbo_tasks(trace_ignore)]
+        stylesheet: StyleSheetLike<'static, 'static>,
+
+        references: Vc<ModuleReferences>,
+
+        url_references: Vc<UnresolvedUrlReferences>,
+
+        #[turbo_tasks(trace_ignore)]
+        exports: Option<IndexMap<String, CssModuleExport>>,
+
+        #[turbo_tasks(trace_ignore)]
+        placeholders: HashMap<String, Url<'static>>,
+
+        #[turbo_tasks(trace_ignore)]
+        options: ParserOptions<'static, 'static>,
+    },
+    Unparseable,
+    NotFound,
+}
+
+impl PartialEq for CssWithPlaceholderResult {
+    fn eq(&self, _: &Self) -> bool {
+        false
+    }
+}
+
+#[turbo_tasks::value(shared, serialization = "none", eq = "manual")]
+pub enum FinalCssResult {
+    Ok {
+        #[turbo_tasks(trace_ignore)]
+        output_code: String,
+
+        #[turbo_tasks(trace_ignore)]
+        exports: Option<CssModuleExports>,
+
+        source_map: Vc<ParseCssResultSourceMap>,
+    },
+    Unparseable,
+    NotFound,
+}
+
+impl PartialEq for FinalCssResult {
+    fn eq(&self, _: &Self) -> bool {
+        false
+    }
+}
+
+#[turbo_tasks::function]
+pub async fn process_css_with_placeholder(
+    result: Vc<ParseCssResult>,
+) -> Result<Vc<CssWithPlaceholderResult>> {
+    let result = result.await?;
+
+    match &*result {
+        ParseCssResult::Ok {
+            cm,
+            stylesheet,
+            references,
+            url_references,
+            options,
+        } => {
+            let stylesheet = stylesheet.to_static(options.clone());
+
+            let (result, _) = stylesheet.to_css(cm.clone(), false, false, false)?;
+
+            let exports = result.exports.map(|exports| {
+                let mut exports = exports.into_iter().collect::<IndexMap<_, _>>();
+
+                exports.sort_keys();
+
+                exports
+            });
+
+            Ok(CssWithPlaceholderResult::Ok {
+                cm: cm.clone(),
+                exports,
+                references: *references,
+                url_references: *url_references,
+                placeholders: HashMap::new(),
+                stylesheet,
+                options: options.clone(),
+            }
+            .into())
```

Comment:
```suggestion
            let (result, _) = stylesheet.to_css(cm.clone(), false, false, false)?;

            let exports = result.exports.map(|exports| {
                let mut exports = exports.into_iter().collect::<IndexMap<_, _>>();

                exports.sort_keys();

                exports
            });

            Ok(CssWithPlaceholderResult::Ok {
                cm: cm.clone(),
                exports,
                references: *references,
                url_references: *url_references,
                placeholders: HashMap::new(),
                options: options.clone(),
                // store Vc<ParseCssResult> instead
            }
            .into())
```

I think we don't need to clone the stylesheet. We can reference it from the original ParseCssResult later instead of storing it in the struct.

Reply:
Done